### DISCUSSION TITLE
documentation on templates

### DISCUSSION THREAD

**Discussion by bluebrown** [aarondl/sqlboiler#1043] [Q&A] [answered]
Hi, is there any material where I could learn more about providing custom templates? I am looking to generate some code that helps to map http query parameters to statements.

**Comment by aarondl** [comment, marked as answer]
There's not too much. Look at the existing templates, override the existing templates with the command line flags and you're done.

**Reply by bluebrown** [reply]
Where are the default templates, I cannot find them in my system. I have used go mod to fetch sqlboiler.  

Once I found the templates, can I put this in my config toml file?

```toml
templates = [
  "/path/to/sqlboiler/templates",
  "/path/to/sqlboiler/templates_test",
  "/path/to/your_project/more_templates"
]
```

**Reply by bluebrown** [reply]
I have found them, but it's a bit inconvenient since I don't know how to reference my home dir from the toml file. ~ and $HOME don't work. So I needed to hardcode `/home/blue/go/pkg/mod/github.com/volatiletech/sqlboiler/v4@v4.8.3/templates/main` which would break on someone else's machine.

**Reply by aarondl** [reply]
The idea is to provide a set of templates that comes from a different repo, not from your specific home directory haha.

One way to do this is by using sqlboiler as a library, you'll see that there's ways to override the embed.FS that's used for the templates. But generally what you'd do is:

In a project called: MyProject, you would include a directory inside there full of custom templates, then you would run sqlboiler from the root of that as intended:

```
MyProject
- templates
- sqlboiler.toml
```

```bash
cd ~/MyProject
sqlboiler --templates ./templates psql
```

Something like that. There's never a reason to point to sqlboiler's templates with the `--templates` flag, the default behavior is those templates are being used.

**Reply by bluebrown** [reply]
OK, thanks. That's what I am doing, but as stated in the docs, I need to include the default templates too.

From  https://github.com/volatiletech/sqlboiler#templates
> Note: Because the --templates flag overrides the embedded templates of sqlboiler, if you still wish to generate the default templates it's recommended that you include the path to sqlboiler's templates as well.

Since I only want to sprinkle some extra stuff onto the existing templates, I did it like that. 

```toml
templates = [
  "/home/blue/go/pkg/mod/github.com/volatiletech/sqlboiler/v4@v4.8.3/templates/main",
  "/home/blue/go/pkg/mod/github.com/volatiletech/sqlboiler/v4@v4.8.3/templates/test",
  "./templates"
]
```

**Reply by aarondl** [reply]
Makes sense. Forgot that was there. This use case isn't that well traveled. I'd probably vendor them in your own directory :)
